plugin/src/main/java/com/denizenscript/denizen/objects/EntityTag.java

```diff
@@ -3159,6 +3158,26 @@ else if (object.getBukkitEntity() instanceof Hanging hanging) {
             }
         });
 
+        // <--[mechanism]
+        // @object EntityTag
+        // @name is_sleeping
+        // @input ElementTag(Boolean)
+        // @description
+        // Sets whether a fox is sleeping.
+        // The entity may wake up immediately after setting this to true. If this is not desired, disable <@link mechanism EntityTag.has_ai>.
+        // @tags
+        // <EntityTag.is_sleeping>
+        // -->
+        tagProcessor.registerMechanism("is_sleeping", false, ElementTag.class, (object, mechanism, input) -> {
+            if (mechanism.requireBoolean()) {
+                if (object.getBukkitEntity() instanceof Fox fox) {
+                    fox.setSleeping(input.asBoolean());
+                    return;
+                }
+                mechanism.echoError("'is_sleeping' mechanism is only valid for Fox entities.");
+            }

```

Comment:
This was a lot cleaner with the early returns you had earlier imo, just have `!requireBoolean -> return`, `!instanceof Fox -> return`, and then the `setSleeping`.
Otherwise LGTM.